mitmproxy/tools/console/commander/commander.py

```diff
@@ -172,6 +214,11 @@ def keypress(self, size, key):
             self.cbuf.left()
         elif key == "right":
             self.cbuf.right()
+        elif key == "up":
+            self.history.add_command(self.cbuf)
+            self.cbuf = self.history.get_prev() or self.cbuf
+        elif key == "down":
+            self.cbuf = self.history.get_next() or self.cbuf
         elif key == "tab":
```

Comment:
No need to do anything about this now, but we're growing a special little garden of hard-coded keys here. We need to add a key binding context for the command editor, make all of these actions into bind-able commands, and then during interaction with the command edit buffer, we first check if a key is bound in this context, and if not, treat it as an ordinary edit action. No need to do that in this PR,  though.